### ISSUE TITLE
computing resources

### ISSUE THREAD

**Issue by Germany321** [MichalZawalski/embodied-CoT#3]
Thanks for releasing your great work. What are the approximate computing resources required to train the model and the time required?

**Comment by betray12138** [comment]
@Germany321 Hello, do you reproduce ECoT with the README guidance? I cannot find the code to resolve Bridge-V2 dataset and the given json reasoning file? Thanks!

**Comment by Everloom-129** [comment]
hi there, thanks for the team's work, it is very innovative and solid!

I noticed the paper said:
> We also ask it to briefly explain the primitive movement and chosen sub-task
in each step, which we include in the ECoT training data. We run our data generation pipeline on the
complete Bridge v2 dataset [13], with more than 2.5M transitions, over the course of 7 days.

I would be interested in the rough estimation of GPU requirement for this tasks, so we can design after-research more clearly. Thanks for the explaination!

**Comment by MichalZawalski** [comment]
> @Germany321 Hello, do you reproduce ECoT with the README guidance? I cannot find the code to resolve Bridge-V2 dataset and the given json reasoning file? Thanks!

@betray12138 Do you have problems with finding those datasets, or with linking them to the training?

Bridge V2 is available at https://rail-berkeley.github.io/bridgedata/, our reasonings at https://huggingface.co/datasets/Embodied-CoT/embodied_features_bridge. To use them, you may update the configuration that you specify in the training command (in https://github.com/MichalZawalski/embodied-CoT/blob/main/prismatic/conf/vla.py), or pass it as an argument in the training command, or simply put in https://github.com/MichalZawalski/embodied-CoT/blob/main/prismatic/vla/datasets/rlds/dataset.py#L60.

**Comment by MichalZawalski** [comment]
> hi there, thanks for the team's work, it is very innovative and solid!
> 
> I noticed the paper said:
> 
> > We also ask it to briefly explain the primitive movement and chosen sub-task
> > in each step, which we include in the ECoT training data. We run our data generation pipeline on the
> > complete Bridge v2 dataset [13], with more than 2.5M transitions, over the course of 7 days.
> 
> I would be interested in the rough estimation of GPU requirement for this tasks, so we can design after-research more clearly. Thanks for the explaination!

Thanks a lot for your kind words! We used 6 to 8 H100s, and the training usually took 4–5 days, actually I think 7 is an upper bound. We followed the criterion employed by OpenVLA: training until the action accuracy reached around 95%. That said, I don't think we've checked whether stopping the training earlier yields similar results, or if training for longer leads to further improvements.

We also noticed that with larger batch sizes the training is faster and usually converges to better results. However, I was not the only user of those GPUs, so not always I could use the largest possible.
